Taggers/plugins/WHLeptonicTagProducer.cc

```diff
@@ -45,366 +46,463 @@ namespace flashgg {
 
             WHLeptonicTag whleptonictags_obj( dipho, mvares );
             whleptonictags_obj.includeWeights( *dipho );
-            
-            if( dipho->leadingPhoton()->pt() < ( dipho->mass() )*leadPhoOverMassThreshold_ ) { continue; }
-            if( dipho->subLeadingPhoton()->pt() < ( dipho->mass() )*subleadPhoOverMassThreshold_ ) { continue; }
-          
+
             idmva1 = dipho->leadingPhoton()->phoIdMvaDWrtVtx( dipho->vtx() );
             idmva2 = dipho->subLeadingPhoton()->phoIdMvaDWrtVtx( dipho->vtx() );
             
-            if( idmva1 <= PhoMVAThreshold_ || idmva2 <= PhoMVAThreshold_ ) { continue; }
-            if( mvares->result < MVAThreshold_ ) { continue; }
+            if( dipho->leadingPhoton()->pt()    < dipho->mass() * leadPhoOverMassThreshold_ ) continue;
+            if( dipho->subLeadingPhoton()->pt() < dipho->mass() * subleadPhoOverMassThreshold_ ) continue;
+            if( idmva1 <= PhoMVAThreshold_ || idmva2 <= PhoMVAThreshold_ ) continue;
+            if( mvares->result < MVAThreshold_ ) continue;
+
+            // Lepton
+            std::vector<edm::Ptr<flashgg::Muon> > goodMuons = selectMuons(  theMuons->ptrs(), 
+                                                                            dipho, 
+                                                                            vertices->ptrs(), 
+                                                                            muonEtaThreshold_, 
+                                                                            leptonPtThreshold_,
+                                                                            muPFIsoSumRelThreshold_, 
+                                                                            deltaRMuonPhoThreshold_, 
+                                                                            deltaRMuonPhoThreshold_ 
+                                                                            );
             
-            photonSelection = true;
-            std::vector<edm::Ptr<flashgg::Muon> > goodMuons = selectMuons( theMuons->ptrs(), dipho, vertices->ptrs(), muonEtaThreshold_, leptonPtThreshold_,
-                    muPFIsoSumRelThreshold_, deltaRMuonPhoThreshold_, deltaRMuonPhoThreshold_ );
-            
-            std::vector<edm::Ptr<Electron> >goodElectrons = selectStdElectrons( theElectrons->ptrs(), dipho,vertices->ptrs(), leptonPtThreshold_, electronEtaThresholds_,
-                                                                                useElectronMVARecipe_,useElectronLooseID_,
-                                                                                deltaRPhoElectronThreshold_,DeltaRTrkElec_,deltaMassElectronZThreshold_,
-                                                                                rho_, evt.isRealData() );
-            
-            hasGoodElec = ( goodElectrons.size() > 0 );
+            std::vector<edm::Ptr<Electron> >goodElectrons = selectStdElectrons( theElectrons->ptrs(), 
```

Comment:
From which file are you loading this? please check the issue here: 
https://github.com/cms-analysis/flashgg/issues/1195

Reply:
Leptonselection.cc is applied in WH Leptonic Tag and ZH Leptonic Tag.

Reply:
Are you sure those are the recommended IDs from the respective POGs?

Reply:
In LeptonSelection.cc, WH and ZH leptonic tag use selectStdElectrons function which has cut-based loose electron ID called from MIcroAOD (like electron->passLooseId(), I think this is recommended) and photon veto, but LeptonSelection2018.cc uses electron MVA WP so I didn't use it. Maybe these two files should be merged to a single file which can choose cut-based ID or MVA WP.